core/src/main/scala/io/projectglow/gff/GffDataSource.scala

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2019 The Glow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.gff
+
+import io.projectglow.common.FeatureSchemas._
+import io.projectglow.gff.GffDataSource._
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.ScalaReflection
+import org.apache.spark.sql.catalyst.expressions.StringSplit
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.sources.{BaseRelation, DataSourceRegister, Filter, PrunedFilteredScan, RelationProvider, SchemaRelationProvider, TableScan}
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.SQLUtils.structFieldsEqualExceptNullability
+
+
+class GffDataSource extends RelationProvider with SchemaRelationProvider with DataSourceRegister {
+
+  override def shortName(): String = "gff"
+
+  /**
+   * Creates relation
+   *
+   * @param sqlContext spark sql context
+   * @param parameters parameters for job
+   * @return Base relation
+   */
+  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String]): BaseRelation = {
+
+    createRelation(sqlContext, parameters, null)
+  }
+
+  /**
+   * Creates relation with user schema
+   *
+   * @param sqlContext spark sql context
+   * @param parameters parameters for job
+   * @param schema     user defined schema
+   * @return Base relation
+   */
+  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String], schema: StructType): BaseRelation = {
+    val path = parameters.get("path")
+    path match {
+      case Some(p) =>
+        val requiredSchema = if (schema == null) {
+          inferSchema (sqlContext, p)
+        } else {
+          schema
+        }
+
+        new GffResourceRelation(sqlContext, p, requiredSchema)
+      case _ => throw new IllegalArgumentException("Path is required")
+    }
+  }
+
+}
+
+
+class GffResourceRelation(
+  val sqlContext: SQLContext,
+  path: String,
+  requiredSchema: StructType
+) extends BaseRelation with TableScan  { // TODO: investigate use of PrunedFilteredScan
```

Comment:
IMO the column pruning would be helpful and probably very simple to implement. I doubt the filters are worth spending time on.

Reply:
Incorporated ColumnPruning. Skipped Filters